### PR TITLE
chore: Adds Envelop plugin system

### PR DESCRIPTION
**Pull request by danzanzini**
## What's the purpose of this pull request?
This PR adds the [envelop plugin system](https://www.envelop.dev/) to our GraphQL layer. More at: https://vtex-dev.atlassian.net/browse/FSSS-12

## How it works? 
All the <em>enveloping</em> logic is encapsulated at the `server/envelop/index.ts` file. In this file, we may add and configure all plugins we want to add to our GraphQL request. I've already added some **regarding performance and security**.

### Error Handling
As part of this task, a **primitive error handling function was added**. If the error returned from the GraphQL request is a [BadRequestError](https://github.com/vtex/faststore/pull/1034), our server will return 400 instead of the default GraphQL 200. We may improve this solution in the future by supporting other <em>flavors</em> of errors.

## How to test it?
Just enter in the preview URL and see that everything is working fine 😄 

## References
- [Envelop plugin hub](https://www.envelop.dev/plugins)
- [Apollo error handling](https://www.apollographql.com/docs/apollo-server/data/errors/)

### CODE REVIEW

**Review by igorbrasileiro — APPROVED** [review]
Amazin work @danzanzini !!
Just a tip, what do you think about adding the [`compose-middleware` package](https://www.npmjs.com/package/compose-middleware)?

File: src/api/graphql.ts [hunk 1]
```diff
@@ -40,14 +55,16 @@ const handler = async (
     return
   }
 
+  const enveloped = await getEnveloped(maskError)
```
Comment (tlgimenes) [inline src/api/graphql.ts:58]: AFAIK this should be run outside the `handler` function, can you confirm it? According to their docs, this is done outside the request cycle (https://www.envelop.dev/docs/getting-started#use-your-envelop)

File: src/server/envelop/index.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+export const getEnveloped = async (formatError: FormatError) =>
+  envelop({
+    plugins: [
+      useSchema(await getSchema()),
+      useExtendContext(getContextFactory()),
+      enableIf(isProduction, () => useMaskedErrors({ formatError })),
+      useGraphQlJit(),
+      useValidationCache(),
+      useParserCache(),
+      useResponseCache(),
```
Comment (tlgimenes) [inline src/server/envelop/index.ts:31]: 👀 This uses memory cache right? If so, we could remove this, right?

File: src/server/envelop/index.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+const isProduction = ENV === 'production'
+
+export type FormatError = FormatErrorHandler
+
+export const getEnveloped = async (formatError: FormatError) =>
+  envelop({
+    plugins: [
+      useSchema(await getSchema()),
+      useExtendContext(getContextFactory()),
+      enableIf(isProduction, () => useMaskedErrors({ formatError })),
```
Comment (tlgimenes) [inline src/server/envelop/index.ts:27]: Isn't it better to always mask the errors and console.log it instead so people don't use error respose on the frontend as logic?

Comment (danzanzini) [reply src/server/envelop/index.ts]: Makes sense. The error's already being logged at dev 👍

**Comment by danzanzini** [comment]
Thanks, @igorbrasileiro. I've added a new Jira task to accomplish this later.

@tlgimenes, I've tackled all your comments. The PR is ready to be reviewed again.

**Review by tlgimenes — APPROVED** [review]
LGTM 🎉

**Pull request merged**